Aggiunta sezione Istruzioni per l'uso (#182)
SevenBitsSwe/7BitsDocs: RTB/Manuale_Utente.tex

\usepackage[utf8]{inputenc}
\usepackage{tabularx}
\usepackage{hyperref}
\usepackage{array}
\usepackage{graphicx}
\usepackage{geometry}
\usepackage{fancyhdr}
\usepackage{tikz}
\usepackage{ragged2e}
\usepackage{anyfontsize}
\usepackage[table,xcdraw]{xcolor}
\usepackage{tabularx, etoolbox}
\usepackage{eso-pic}
\usepackage{titlesec}
\usepackage{float}
\usepackage{longtable}

> \newcommand\version{1.0.4} %aggiunta versione come variabile

Reviewer: ```suggestion
\newcommand\version{0.1.4} %aggiunta versione come variabile
```

Ogni pannello è progettato per presentare i dati in modo ottimale in base alla loro tipologia. In particolare, per le informazioni di natura geospaziale, viene utilizzata una mappa che mostra dei marker che rappresentano la posizione dei sensori, delle attività e dei messaggi pubblicitari, insieme ai relativi dettagli. All'interno della dashboard principale, è inoltre presente un pannello tabellare che fornisce un report dettagliato delle attività, aggregandole in base al numero di messaggi generati nell'ultimo mese. Questo consente di ottenere una panoramica chiara e strutturata sull'andamento del sistema.\\
Oltre alla dashboard principale, gli utenti autorizzati possono accedere alle dashboard specifiche degli utenti standard, ciascuna composta da un pannello dedicato alla visualizzazione della mappa, che include marker per ogni attività e si concentra sull'intero percorso svolto e su tutti i messaggi ricevuti dal singolo utente.\\
\begin{figure}[H]
    \centering
    \includegraphics[width=0.1\linewidth]{LogoUnipd.png}
    \caption{Dashboard principale}
\end{figure}

    \subsubsection{Barra degli strumenti} % pulsante edit mode da tenere?

>     La barra degli strumenti, situata nella parte superiore della dashboard, offre una serie di opzioni aggiuntive tramite pulsanti e menu a tendina. Questi consentono di eseguire azioni specifiche e accedere a funzionalità aggiuntive in modo semplice e rapido.\\

Reviewer: ```suggestion
    La barra degli strumenti, situata nella parte superiore della dashboard, offre una serie di opzioni aggiuntive tramite pulsanti e menù a tendina, che consentono di eseguire azioni specifiche e accedere a funzionalità aggiuntive in modo semplice e rapido.\\
```

    \begin{figure}[H]
    \centering
    \includegraphics[width=0.1\linewidth]{LogoUnipd.png}
    \caption{Barra degli strumenti}
    \end{figure}

    \paragraph{Pulsante profilo utente}

>     Il pulsante del "profilo utente" consente di accedere rapidamente alle impostazioni personali, dove è possibile visualizzare e modificare le informazioni del proprio account. Premendo il pulsante, viene visualizzato un menu a tendina con le seguenti opzioni:

Reviewer: ```suggestion
    Il pulsante del "profilo utente" consente di accedere rapidamente alle impostazioni personali, dove è possibile visualizzare e modificare le informazioni del proprio account. Premendo il pulsante, viene visualizzato un menù a tendina con le seguenti opzioni:
```

    \begin{itemize}
        \item \textbf{Profile}: permette di consultare e aggiornare le informazioni personali;
        \item \textbf{Notification history}: consente di visualizzare la cronologia delle notifiche ricevute;
        \item \textbf{Change password}: offre la possibilità di modificare la password dell'account;
        \item \textbf{Sign out}: consente di effettuare il logout dal sistema.
    \end{itemize}
    \begin{figure}[H]
    \centering
    \includegraphics[width=0.1\linewidth]{LogoUnipd.png}
    \caption{Pulsante profilo utente}
    \end{figure}

    \paragraph{Barra di ricerca}
    La barra di ricerca consente agli utenti autorizzati di cercare e filtrare rapidamente le dashboard disponibili o di navigare all'interno del sistema. È possibile digitare il nome della dashboard o della pagina desiderata e premere "Invio" per visualizzare i risultati corrispondenti.
    \begin{figure}[H]
    \centering
    \includegraphics[width=0.1\linewidth]{LogoUnipd.png}
    \caption{Barra di ricerca}
    \end{figure}

>     \paragraph{Menu}

Reviewer: ```suggestion
    \paragraph{Menù}
```

>     Il menu, accessibile tramite l'icona di Grafana situata sulla barra degli strumenti, offre agli utenti autorizzati l'accesso a diverse funzionalità aggiuntive. Le opzioni disponibili sono le seguenti:

Reviewer: ```suggestion
    Il menù, accessibile tramite l'icona di Grafana situata sulla barra degli strumenti, offre agli utenti autorizzati l'accesso a diverse funzionalità aggiuntive. Le opzioni disponibili sono le seguenti:
```

        \item \textbf{Dashboards}: consente di accedere alla lista completa delle dashboard disponibili;
        \item \textbf{Explore}: offre strumenti per esplorare i dati e analizzarli in tempo reale;
        \item \textbf{Altering}: permette di visualizzare, gestire ed esportare le regole di allerta e le notifiche;
        \item \textbf{Connections}: consente di gestire le connessioni a database e fonti dati esterne;
        \item \textbf{Administration}: fornisce accesso alle impostazioni avanzate e alla gestione degli utenti e dei permessi.
    \end{itemize}
    \begin{figure}[H]
    \centering
    \includegraphics[width=0.1\linewidth]{LogoUnipd.png}
    \caption{Menu}
    \end{figure}

    \paragraph{Breadcrumb}
    La breadcrumb mostra il percorso di navigazione dell'utente autorizzato e indica la sua posizione corrente all'interno del sistema attraverso una serie di link. Questi link consentono di navigare facilmente tra le sezioni, permettendo all'utente di tornare rapidamente alle pagine precedenti.
    \begin{figure}[H]
    \centering
    \includegraphics[width=0.1\linewidth]{LogoUnipd.png}
    \caption{Breadcrumb}
    \end{figure}

>     \paragraph{Pulsante mark as favorite}

Reviewer: ```suggestion
    \paragraph{Pulsante Mark as favorite}
```

>     Il pulsante "Mark as favorite", consente di aggiungere una dashboard all'elenco delle preferite, facilitandone l'accesso rapido.

Reviewer: ```suggestion
    Il pulsante "Mark as favorite" consente di aggiungere una dashboard all'elenco delle preferite, facilitandone l'accesso rapido.
```

    \begin{figure}[H]
    \centering
    \includegraphics[width=0.1\linewidth]{LogoUnipd.png}
    \caption{Pulsante "Mark as favorite"}
    \end{figure}

>     \paragraph{Pulsante share}

Reviewer: ```suggestion
    \paragraph{Pulsante Share}
```

>     Il pulsante "Share", consente di condividere la dashboard attualmente visualizzata. Cliccando sulla freccia rivolta verso il basso accanto al pulsante, si apre un menu con le seguenti opzioni:

Reviewer: ```suggestion
    Il pulsante "Share", consente di condividere la dashboard attualmente visualizzata. Cliccando sulla freccia rivolta verso il basso accanto al pulsante, si apre un menù con le seguenti opzioni:
```

    \begin{itemize}

>         \item \textbf{Share internally}: per condividere la dashboard con altri utenti all'interno dell'organizzazione, mantenendo il controllo sugli accessi e sui permessi.

Reviewer: ```suggestion
        \item \textbf{Share internally}: per condividere la dashboard con altri utenti all'interno dell'organizzazione, mantenendo il controllo sugli accessi e sui permessi;
```

>         \item \textbf{Share externally}: per rendere la dashboard accessibile pubblicamente, generando un link che chiunque può utilizzare per visualizzare la dashboard in modalità di sola lettura.

Reviewer: ```suggestion
        \item \textbf{Share externally}: per rendere la dashboard accessibile pubblicamente, generando un link che chiunque può utilizzare per visualizzare la dashboard in modalità di sola lettura;
```

        \item \textbf{Share snapshot}: per creare uno snapshot della dashboard da condividere. Questo metodo consente di condividere lo stato attuale della dashboard, rimuovendo dati sensibili come query e link ai pannelli.
    \end{itemize}
    \begin{figure}[H]
    \centering
    \includegraphics[width=0.1\linewidth]{LogoUnipd.png}
    \caption{Pulsante "Share"}
    \end{figure}

    \paragraph{Time settings}

>     Il menu a tendina "Time settings" permette all'utente di selezionare l'intervallo temporale desiderato per la visualizzazione dei dati. La dashboard offre una serie di intervalli predefiniti e consente all'utente di personalizzarli secondo le proprie necessità. Le aggregazioni delle misurazioni e le operazioni sui dati vengono applicate in base all'intervallo temporale selezionato, garantendo un'analisi accurata e contestualizzata delle informazioni.

Reviewer: ```suggestion
    Il menù a tendina "Time settings" permette all'utente di selezionare l'intervallo temporale desiderato per la visualizzazione dei dati. La dashboard offre una serie di intervalli predefiniti e consente all'utente di personalizzarli secondo le proprie necessità. Le aggregazioni delle misurazioni e le operazioni sui dati vengono applicate in base all'intervallo temporale selezionato, garantendo un'analisi accurata e contestualizzata delle informazioni.
```

    \begin{figure}[H]
    \centering
    \includegraphics[width=0.1\linewidth]{LogoUnipd.png}

>     \caption{Menu "Time settings"}

Reviewer: ```suggestion
    \caption{Menù "Time settings"}
```

    \end{figure}

>     \paragraph{Pulsante zoom out}

Reviewer: ```suggestion
    \paragraph{Pulsante Zoom out}
```

    Il pulsante "Zoom out" consente di estendere l'intervallo temporale visualizzato nei grafici, permettendo di analizzare periodi più ampi di dati. Cliccando su questo pulsante, la visualizzazione si adatta automaticamente per mostrare un intervallo temporale più esteso, secondo gli intervalli predefiniti.
    \begin{figure}[H]
    \centering
    \includegraphics[width=0.1\linewidth]{LogoUnipd.png}
    \caption{Pulsante "Zoom out"}
    \end{figure}
    

>     \paragraph{Pulsante refresh}

Reviewer: ```suggestion
    \paragraph{Pulsante Refresh}
```

>     Il pulsante "Refresh" consente all'utente di aggiornare la dashboard attualmente visualizzata per ottenere i dati più recenti. Cliccando sulla freccia accanto al pulsante, si apre un menu a tendina che offre diverse opzioni per la frequenza di aggiornamento:

Reviewer: ```suggestion
    Il pulsante "Refresh" consente all'utente di aggiornare la dashboard attualmente visualizzata per ottenere i dati più recenti. Cliccando sulla freccia accanto al pulsante, si apre un menù a tendina che offre diverse opzioni per la frequenza di aggiornamento:
```

        \item \textbf{Intervalli predefiniti}: permettono di impostare un aggiornamento automatico a intervalli specifici (ad esempio, 5s, 10s, 30s, 1m, ecc.).
    \end{itemize}
    Di default, le dashboard del sistema sono configurate per aggiornarsi automaticamente ogni 10 secondi. Questa impostazione può essere modificata in base alle esigenze dell'utente autorizzato.
    \begin{figure}[H]
    \centering
    \includegraphics[width=0.1\linewidth]{LogoUnipd.png}
    \caption{Pulsante "Refresh"}
    \end{figure}

    \subsubsection{Pannelli}
    I pannelli costituiscono la componente principale della dashboard, presentando i dati in modo interattivo. Ogni pannello è un widget progettato per visualizzare un insieme specifico di dati, utilizzando una forma appropriata in base alla loro tipologia. Nel contesto del sistema in questione, i pannelli includono una mappa generale, delle mappe specifiche per utenti e una tabella riepilogativa dei punti di interesse.\\
    Gli elementi presenti nei pannelli sono:
    \begin{itemize}
        \item Nome del pannello;
        \item Legenda (se presente);

>         \item Menu opzioni del pannello;

Reviewer: ```suggestion
        \item Menù opzioni del pannello;
```

        \item Visualizzazione dei dati.
    \end{itemize}

    \paragraph{Legenda}
    La legenda è una sezione del pannello che fornisce la chiave di lettura dei dati visualizzati. È utile per comprendere il significato dei colori e delle forme utilizzate nel pannello. Viene impiegata solamente nei pannelli che contengono una mappa.
    \begin{figure}[H]
    \centering
    \includegraphics[width=0.1\linewidth]{LogoUnipd.png}
    \caption{Legenda}
    \end{figure}

>     \paragraph{Menu Opzioni}

Reviewer: ```suggestion
    \paragraph{Menù Opzioni}
```

    Cliccando sull'icona con tre puntini verticali nell'angolo in alto a destra di un pannello, è possibile accedere a diverse funzionalità, tra cui la visualizzazione a schermo intero, la condivisione o l'esportazione del pannello, l'esplorazione delle query per un'analisi più approfondita, l'ispezione dei dati visualizzati e l'accesso a opzioni aggiuntive.
    \begin{figure}[H]
    \centering
    \includegraphics[width=0.1\linewidth]{LogoUnipd.png}
    \caption{Opzioni del pannello}
    \end{figure}

\subsection{Struttura e funzionalità dei pannelli}

> Questa sezione descrive nel dettaglio i pannelli principali che compongono le dashboard del sistema, fornendo una panoramica delle funzionalità e delle informazioni visualizzate. In particolare, vengono illustrate la Mappa generale, la Mappa specifica e la Tabella attività.

Reviewer: ```suggestion
Questa sezione descrive nel dettaglio i pannelli principali che compongono le dashboard del sistema, fornendo una panoramica delle funzionalità e delle informazioni visualizzate. In particolare, sono disponibili tre tipi di dashboard, descritti di seguito.
```

    \subsubsection{Mappa generale}

>     La mappa principale presenta una visualizzazione globale dei dati geospaziali, utilizzando dei marker distinti per rappresentare diverse entità e attività. I punti di interesse sono indicati da marker rossi immobili, che evidenziano le attività che offrono dei servizi. I sensori, rappresentati da marker verdi in movimento, seguono gli spostamenti degli utenti sulla mappa, fornendo dati in tempo reale sulla loro ultima posizione. I messaggi pubblicitari, mostrati come quadrati arancioni, indicano l'ultimo messaggio pubblicitario generato per ciascun utente; questi sono visibili soltanto se l'utente si trova all'interno dell'area di influenza dell'attività correlata.\\ 

Reviewer: i tre tipi di marker con descrizione li inserirei come elenco puntato per migliorare la leggibilità

>     La mappa include una leggenda che facilita l'interpretazione dei marker, associando colori e forme alle diverse categorie di dati. È possibile navigare sulla mappa con il mouse mentre lo zoom può essere regolato utilizzando la rotella del mouse oppure i pulsanti "+" e "-" situati sotto al nome del pannello.

Reviewer: ```suggestion
    La mappa include una leggenda che facilita l'interpretazione dei marker, associando colori e forme alle diverse categorie di dati. È possibile navigare sulla mappa con il mouse mentre lo zoom può essere regolato utilizzando la rotella del mouse oppure tramite i pulsanti "+" e "-", situati sotto al nome del pannello.
```

    \begin{figure}[H]
    \centering
    \includegraphics[width=0.1\linewidth]{LogoUnipd.png}
    \caption{Pannello mappa generale}
    \end{figure}

    \subsubsection{Mappa specifica}

>     La mappa specifica presenta una visualizzazione globale dei dati geospaziali relativi a un singolo utente, utilizzando dei marker distinti per rappresentare diverse entità e attività. I punti di interesse sono indicati da marker rossi immobili, che evidenziano le attività che offrono dei servizi. Il percorso dell'utente è rappresentato da una serie di marker viola che tracciano tutte le posizioni percorse, con il primo e l'ultimo punto evidenziati da un contorno verde. I messaggi pubblicitari, mostrati come quadrati arancioni, indicano tutti i messaggi pubblicitari ricevuti dall'utente, ciascuno situato sulla posizione esatta in cui l'utente si trovava quando lo ha ricevuto.\\

Reviewer: lista marker come commento sopra

>     La mappa include una leggenda che facilita l'interpretazione dei marker. È possibile navigare sulla mappa con il mouse mentre lo zoom può essere regolato utilizzando la rotella del mouse oppure i pulsanti "+" e "-" situati sotto al nome del pannello.

Reviewer: ```suggestion
    La mappa include una leggenda che facilita l'interpretazione dei marker. È possibile navigare sulla mappa con il mouse mentre lo zoom può essere regolato utilizzando la rotella del mouse oppure tramite i pulsanti "+" e "-", situati sotto al nome del pannello.
```
